src/DGtal/helpers/Shortcuts.h

```diff
@@ -1373,6 +1373,68 @@ namespace DGtal
       }    
 
     
+      /// Given any digital surface, returns the vector of its k-dimensional cells.
+      ///
+      /// @note The order of cells is given by the default traversal
+      /// of the surfels of the surface, where the cells of each
+      /// surfel are visited in no specific order.
+      ///
+      /// @tparam TDigitalSurfaceContainer either kind of DigitalSurfaceContainer
+      ///
+      /// @param[out] c2i the map Cell -> Cell index in the cell range.
+      /// @param[in] surface a smart pointer on a (light or not) digital surface (e.g. DigitalSurface or LightDigitalSurface).
+      /// @param[in] k the dimension of the output cells
+      ///
+      /// @return a range of cells as a vector.
+      template <typename TDigitalSurfaceContainer>
+      static CellRange
+      getCellRange
+      ( Cell2Index& c2i,
+        CountedPtr< ::DGtal::DigitalSurface<TDigitalSurfaceContainer> > surface,
+        const Dimension k )
+      {
+        CellRange result;
+        result.reserve( 2 * surface->size() + 100 );
```

Comment:
where does the +100 come from?

Reply:
well.. the overall formula BTW

Reply:
Approximate number of surfels for a 2d surface. I have put a comment in the code